x/cscalist/keeper/grpc_query.go

```diff
@@ -0,0 +1,52 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/rarimo/rarimo-core/x/cscalist/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ types.QueryServer = Keeper{}
+
+func (k Keeper) Params(goCtx context.Context, _ *types.QueryParamsRequest) (*types.QueryParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	return &types.QueryParamsResponse{Params: k.GetParams(ctx)}, nil
+}
+
+func (k Keeper) Tree(c context.Context, req *types.QueryTreeRequest) (*types.QueryTreeResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	optCap := req.Pagination.Limit
+	// prevent extreme memory usage on bigger numbers, while the storage shouldn't
+	// contain more than that
+	if optCap > 600 {
+		optCap = 600
```

Comment:
should we change this value also in pagination parameters?

Reply:
nice catch 🙌